pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -73,17 +60,7 @@ message ConnectionRequest {
     
     string network_service_name = 3;
 
-    repeated MechanismType type = 4;
-
-    KernelInterface kernel_interface = 6;
-
-    VhostUserInterface vhost_user_interface = 7;
-
-    MemInterface mem_interface = 8;
-
-    SRIOV sriov_interface = 9;
-
-    HWInterface hw_interface = 10;
+    repeated NetworkInterface network_interface = 4;
```

Comment:
I do not think it will work well because these would allow to send to a server a list of millions of interfaces and kill the server. 
Second issue: Oneof used for NetworkInterface  message just control which values are acceptable just for this type of variable, but since it is repeated, we can  end up with a request for 3 or 5 or 10 kernel_interfaces.